pkg/handler/scale_handler.go

```diff
@@ -259,18 +258,19 @@ func (h *ScaleHandler) scaleDeployment(deployment *apps_v1.Deployment, scaledObj
 		// Update LastActiveTime to now.
 		now := meta_v1.Now()
 		scaledObject.Status.LastActiveTime = &now
-		h.updateScaledObject(scaledObject)
+		scaledObject, _ = h.updateScaledObject(scaledObject)
 	} else {
 		log.Debugf("deployment (%s/%s) no change", deployment.GetNamespace(), deployment.GetName())
 	}
+	return scaledObject
 }
 
-func (h *ScaleHandler) updateScaledObject(scaledObject *keda_v1alpha1.ScaledObject) error {
-	_, err := h.kedaClient.KedaV1alpha1().ScaledObjects(scaledObject.GetNamespace()).Update(scaledObject)
+func (h *ScaleHandler) updateScaledObject(scaledObject *keda_v1alpha1.ScaledObject) (*v1alpha1.ScaledObject, error) {
+	newscaledObject, err := h.kedaClient.KedaV1alpha1().ScaledObjects(scaledObject.GetNamespace()).Update(scaledObject)
```

Comment:
If you assign the ```scaledObject``` parameter to the output of ```.Update()```, this is the only change needed as ```scaledObject``` is passed as a pointer and thus mutate its value without needing to explicitly return it from the all methods preceding ```updateScaledObject```.

Reply:
Done